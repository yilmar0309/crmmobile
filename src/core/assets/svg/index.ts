export {ArrowBackIcon} from './ArrowBackIcon';
export {CrossFilledIcon} from './CrossFilledIcon';
export {InfoIconOutlined} from './InfoIconOutlined';
