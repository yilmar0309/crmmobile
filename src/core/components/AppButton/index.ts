export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
