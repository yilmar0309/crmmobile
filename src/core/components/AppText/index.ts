export {AppText, AppTextVariant} from './AppText';
