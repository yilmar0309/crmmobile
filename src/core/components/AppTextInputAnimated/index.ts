export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
