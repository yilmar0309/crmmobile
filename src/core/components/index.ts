export {AppText, AppTextVariant} from './AppText';
export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
export {AppContainer} from './AppContainer';
export {AppSpacer} from './AppSpacer';
export {AppHeaderLeftBack} from './AppHeaderLeftBack';
export {AppShowEmptyList} from './AppShowEmptyList';
export {AppLoading} from './AppLoading';
export {AppImage} from './AppImage';
export {AppLoadingList} from './AppLoadingList';
