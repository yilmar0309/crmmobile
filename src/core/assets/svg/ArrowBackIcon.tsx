import * as React from 'react';
import Svg, {Path, SvgProps} from 'react-native-svg';
import {colorsLight} from '@/core/theme';

interface Props extends SvgProps {
  color?: string;
}

export const ArrowBackIcon = ({
  color = colorsLight.CONTENT_SECONDARY,
  ...props
}: Props) => (
  <Svg
    width={props.width ?? 20}
    height={props.height ?? 20}
    fill="none"
    {...props}
    viewBox="0 0 24 24">
    <Path
      fill={color}
      d="m10.8 12l3.9 3.9q.275.275.275.7t-.275.7q-.275.275-.7.275t-.7-.275l-4.6-4.6q-.15-.15-.212-.325T8.425 12q0-.2.063-.375T8.7 11.3l4.6-4.6q.275-.275.7-.275t.7.275q.275.275.275.7t-.275.7z"
    />
  </Svg>
);
