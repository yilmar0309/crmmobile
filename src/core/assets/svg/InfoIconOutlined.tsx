import * as React from 'react';
import Svg, {Path, SvgProps} from 'react-native-svg';
import {colorsLight} from '@/core/theme';

interface Props extends SvgProps {
  color?: string;
}

export const InfoIconOutlined = ({
  color = colorsLight.CONTENT_SECONDARY,
  ...props
}: Props) => (
  <Svg
    width={props.width ?? '24'}
    height={props.height ?? '24'}
    fill="none"
    viewBox="0 0 24 24"
    {...props}>
    <Path
      fill={color ?? '#000'}
      d="M11 7h2v2h-2V7zm0 4h2v6h-2v-6zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
    />
  </Svg>
);
