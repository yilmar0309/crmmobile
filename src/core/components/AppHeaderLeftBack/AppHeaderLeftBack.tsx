import React, {ReactNode} from 'react';
import {StyleSheet, ViewStyle} from 'react-native';
import {TouchableOpacity} from 'react-native-ui-lib';
import {ArrowBackIcon} from '@/core/assets/svg';
import {colorsLight} from '@/core/theme';

interface IAppHeaderLeftBack {
  color?: string;
  onPress?: () => void;
  customIcon?: ReactNode;
  container?: ViewStyle;
}

export const AppHeaderLeftBack = ({
  color = colorsLight.CONTENT_PRIMARY,
  onPress,
  customIcon = <ArrowBackIcon width={44} height={44} color={color} />,
  container = styles.container,
}: IAppHeaderLeftBack) => {
  return (
    <TouchableOpacity
      center
      style={container}
      onPress={onPress}
      testID="header-back-button">
      {customIcon}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingRight: 12,
    paddingLeft: 2,
    paddingVertical: 12,
  },
});
