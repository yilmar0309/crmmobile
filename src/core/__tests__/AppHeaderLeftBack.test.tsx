import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {ArrowBackIcon} from '@/core/assets/svg';
import {AppHeaderLeftBack} from '../components/AppHeaderLeftBack';

describe('AppHeaderLeftBack', () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default props', () => {
    const {getByTestId, toJSON} = render(
      <AppHeaderLeftBack onPress={mockOnPress} />,
    );
    const button = getByTestId('header-back-button');
    expect(button).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    const {getByTestId} = render(<AppHeaderLeftBack onPress={mockOnPress} />);
    const button = getByTestId('header-back-button');
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('renders custom icon when provided', () => {
    const customIcon = (
      <ArrowBackIcon width={30} height={30} color="red" testID="custom-icon" />
    );
    const {getByTestId} = render(
      <AppHeaderLeftBack customIcon={customIcon} onPress={mockOnPress} />,
    );
    const icon = getByTestId('custom-icon');
    expect(icon).toBeTruthy();
    expect(icon.props.width).toEqual(30);
    expect(icon.props.height).toEqual(30);
  });
});
