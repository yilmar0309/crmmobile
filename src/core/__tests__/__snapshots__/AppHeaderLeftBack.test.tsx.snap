// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeaderLeftBack renders correctly with default props 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 2,
      "paddingRight": 12,
      "paddingVertical": 12,
    }
  }
  testID="header-back-button"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={44}
    bbWidth={44}
    fill="none"
    focusable={false}
    height={44}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 44,
          "width": 44,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={44}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="m10.8 12l3.9 3.9q.275.275.275.7t-.275.7q-.275.275-.7.275t-.7-.275l-4.6-4.6q-.15-.15-.212-.325T8.425 12q0-.2.063-.375T8.7 11.3l4.6-4.6q.275-.275.7-.275t.7.275q.275.275.275.7t-.275.7z"
        fill={
          {
            "payload": 4279835675,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
